hail/src/test/scala/is/hail/expr/types/physical/PSubsetStructSuite.scala

```diff
@@ -47,4 +47,15 @@ class PSubsetStructSuite extends HailSuite {
 
     assert(rtV(0)  == viewV(0) && rtV(2) == viewV(1))
   }
+
+  @Test def testConstruction(): Unit = {
```

Comment:
let me know if you want something more comprehensive. I didn't want to spend much time on theoretical tests when so many of our integration tests are available to solve :D